packages/use-intl/src/useTranslations.tsx

```diff
@@ -103,16 +102,7 @@ export default function useTranslations<
   ): any;
 } {
   const context = useIntlContext();
-
   const messages = context.messages as IntlMessages;
-  if (!messages) {
-    const intlError = new IntlError(
-      IntlErrorCode.MISSING_MESSAGE,
-      __DEV__ ? `No messages were configured on the provider.` : undefined
-    );
-    context.onError(intlError);
-    throw intlError;
-  }
```

Comment:
Seems like this was not necessary in the end when we annotated the return type explicitly.